Refactor/color variables

kwajiehao opened the pull request:
### Overview 
This PR refactors the color variables as Jekyll variables which take reference from the front matter in the `theme-colors/custom.scss' file

kwajiehao committed:
refactor: store colors as jekyll variables

prestonlimlianjie reviewed (changes requested):
Will this work for sites that have 10 resource-room categories? Perhaps a better way of doing this is to do the following:

#### Create a `site-colors` array object in `config.yml`
```
site-colors:
- [color-1]
- [color-2]
- [color-3]
- [color-4]
```

#### Template the `site-colors` in `custom.scss`

And subsequently write Jekyll code into `custom.scss`, with the following pseudocode:

```
// There are other SCSS chunks such as .bg-media-color-NUMBER and .bg-NUMBER

for color in site.config.site-colors:
    insert <chunk_for_bg-media-color-NUMBER> templated with color
    insert <chunk_for_bg-NUMBER> templated with color
```

kwajiehao committed:
Refactor: retrieve colors from _config.yml

This commit modifies both _config.yml and custom.scss so that
colors in custom.scss are retrieved programmatically from
_config.yml

kwajiehao commented:
> Will this work for sites that have 10 resource-room categories? Perhaps a better way of doing this is to do the following:
> 
> #### Create a `site-colors` array object in `config.yml`
> ```
> site-colors:
> - [color-1]
> - [color-2]
> - [color-3]
> - [color-4]
> ```
> 
> #### Template the `site-colors` in `custom.scss`
> And subsequently write Jekyll code into `custom.scss`, with the following pseudocode:
> 
> ```
> // There are other SCSS chunks such as .bg-media-color-NUMBER and .bg-NUMBER
> 
> for color in site.config.site-colors:
>     insert <chunk_for_bg-media-color-NUMBER> templated with color
>     insert <chunk_for_bg-NUMBER> templated with color
> ```

I can implement the first part by putting colors in `_config.yml` but it seems that there are two separate issues:
1. How we retrieve colors
2. How many resource colors we're allowing by default



> Will this work for sites that have 10 resource-room categories? Perhaps a better way of doing this is to do the following:
> 
> #### Create a `site-colors` array object in `config.yml`
> ```
> site-colors:
> - [color-1]
> - [color-2]
> - [color-3]
> - [color-4]
> ```
> 
> #### Template the `site-colors` in `custom.scss`
> And subsequently write Jekyll code into `custom.scss`, with the following pseudocode:
> 
> ```
> // There are other SCSS chunks such as .bg-media-color-NUMBER and .bg-NUMBER
> 
> for color in site.config.site-colors:
>     insert <chunk_for_bg-media-color-NUMBER> templated with color
>     insert <chunk_for_bg-NUMBER> templated with color
> ```

Fixed so that colors are retrieved programmatically from `_config.yml`

prestonlimlianjie commented on isomerpages-base/misc/custom.scss:
```diff
@@ -4,16 +4,14 @@
-$media-color-three: #00838f;
-$media-color-four: #00838f;
-$media-color-five: #00838f;
+{%- for media-color in site.colors.media-colors -%}
```
We need to do this for lines 41 to 72 as well.

kwajiehao replied:
done

kwajiehao committed:
refactor: converted bg css to use liquid for loop

prestonlimlianjie reviewed (approved):
lgtm
